azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -180,6 +197,9 @@ def run(self, goal_state, ext_gs_updated):
             self._updater.retrieve_agent_version(agent_family, goal_state)
 
             if not self._updater.is_retrieved_version_allowed_to_update(agent_family):
+                # Reset the last error report msg  when this condition met as there is no issue with update since requested
+                # version is same as current version or below than daemon version. In that case, we shouldn't report any error msg in the status report
```

Comment:
This comment makes it sound like 'is_retrieved_version_allowed_to_update' is only False if requested version is same as current version or less than daemon version. We could also hit this condition if agent_family.is_version_from_rsm is False (RSM Update) or if update is not allowed due to upgrade frequency (self update). 

Logic makes sense to me, but comment is a little confusing

Reply:
Since to me more like we need to reset it on a new goal state. 

@nagworld9 CRP needs this message for the goal state requesting the update, but does it still need it for other goal states? If it does, we are losing it on service restart, since it's only in-memory.

Reply:
I think you are answering my question in the next comment below.

However, we should not rely on that and just stop sending it.

`In other times, we are ok send cache value since nobody looking at that status (not even crp)
`

Reply:
makes sense, we should reset on new goal state and we should send last report msg until new goal state